learnify/backend/api/controllers/learning_space/put_edit_comment.js

```diff
@@ -0,0 +1,83 @@
+
+import { LearningSpace } from '../../../models/index.js';
+import jwt from "jsonwebtoken";
+import { validateEditComment_init } from '../../validators/learning_space_init_validator.js';
+
+
+
+export default async (req, res) => {
+    var username;
+    try {
+        const authHeader = req.headers.authorization;
+        console.log(jwt.decode(authHeader))
+        username = jwt.decode(authHeader).username;
+    } catch (e) {
+        return res.status(401).json({ "resultMessage": "There is something wrong with your auth token." });
+    }
+
+    const { error } = validateEditComment_init(req.body);
+    if (error) {
+        console.log(error);
+        return res.status(400).json({ "resultMessage": "Please check your inputs." });
+    }
+
+    let ls = await LearningSpace.findOne({ _id: req.body.ls_id })
+        .catch((err) => {
+            console.log(err.message);
+            return res.status(500).json({ "resultMessage": "Something is wrong." });
+        });
+    
+    var post;
+    var postIndex = 0;
+        ls.posts.forEach(function(check) {
```

Comment:
I believe if ls returns undefined from findOne() without error, the ls.posts.forEach() will give error for undefined - reading ('forEach'). Correct me if I'm wrong please.

Reply:
I appreciate your review and I think you are right that was my bad. I guess it's complete now.